db/workflow.go

```diff
@@ -73,15 +73,13 @@ func insertInWorkflow(ctx context.Context, db *sql.DB, wf Workflow, tx *sql.Tx)
 }
 
 func insertIntoWfWorkerTable(ctx context.Context, db *sql.DB, wfID uuid.UUID, workerID uuid.UUID, tx *sql.Tx) error {
-	// TODO This command is not 100% reliable for concurrent write operations
 	_, err := tx.Exec(`
 	INSERT INTO
 		workflow_worker_map (workflow_id, worker_id)
-	SELECT $1, $2
-	WHERE
-		NOT EXISTS (
-			SELECT workflow_id FROM workflow_worker_map WHERE workflow_id = $1 AND worker_id = $2
-		);
+	VALUES
+	        ($1, $2)
+	ON CONFLICT (workflow_id, worker_id)
+	DO NOTHING;
```

Comment:
this will still return an error right?

Reply:
No, this will not return error on conflict.

Reply:
This is not what we want right? We should be returning an error so that we can bubble it up to the caller.

Reply:
@mmlb, This actually is what we want. If the row is already in the table, nothing need to do. This behavior is the same as the the original `SELECT ...` code try to achieve. The fix here is to remove `SELECT ...` block, which voilate serializable isolation policy.

Reply:
ah true that makes sense to keep the same behavior. Not sure the original behavior was the best idea, but that should be handled separately if necessary.